### PR TITLE
Decouple from active platform

### PR DESCRIPTION
**Pull request by todi1856**
### Purpose of PR
**Type of change:**
- [X] Improvement

* **Description of the feature**
Removed limitation where Android logcat could be used only when active platform is Android. Android support is still requires to be installed.
* **Links to screenshots, design docs, user docs, etc.**
* **Links to Jira/Fogbugz cases**
https://github.com/Unity-Technologies/com.unity.mobile.logcat/issues/111
* **Halo effect**
n/a
* **Performance Impact**
n/a
* **Package requirements**
2019.2 and up

### Checklist for PR maker
- [ n/a] Have you added a backport label? (if needed)
- [X] Have you updated the Changelog? _Each package has a `CHANGELOG.md` file_
- [X] **Have you added or updated the Documentation to your PR?**

---
### Testing status
* **Existing or new automation tests - what automation was added, changed**
https://yamato.cds.internal.unity3d.com/jobs/277-com.unity.mobile.android-logcat/tree/decouple-from-active-platform/.yamato%252Fupm-ci.yml%2523test_all_trigger
* **Detailed description of what testing was done, what projects were run**
Besides automated tests, did the following manual test:
   * Installed 2019.4 with Android Support
   * Tried using Android Logcat with both active platform Android and Windows Standalone
   * Opened every possible window, spawned from logcat (Stacktrace Utility, Internal Log, Capture Screenshot, etc)
   * Saved layout
   * Quit editor
   * Renamed in <UnityInstall>2019.4.8f1\Editor\Data\PlaybackEngines\AndroidPlayer UnityEditor.Android.Extensions.dll to UnityEditor.Android.Extensions.dll__, 
   * Started Editor
   * Restored layout
   * Checked console that there's no errors, and there's a proper error message in Android Logcat window about missing Android Support
   * Restored UnityEditor.Android.Extensions.dl

* **What environment did you test on?**
Windows
* **Projects**
Used https://github.com/Unity-Technologies/com.unity.mobile.logcat/tree/master/TestProjects/SampleProject1

**Juozas> no rush on this PR, just feel free to review when you'll have time**

### CODE REVIEW

**Review by JuozasK — APPROVED** [review]
Looks good! Great testing description! Reviewed your tests + did some of mine, basically ran through the windows as well, checking for anything that might break if Android Platform is not selected.

File: com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+namespace Unity.Android.Logcat
+{
+    /// <summary>
+    /// Provides dynamic way of accessing UnityEditor.Android.Extensions.dll.
+    /// It solves the problem where if you have logcat package installed, but there's no Android Support installed, you won't get compiler errors
+    /// It also enables to use Logcat package when active platform is not Android
+    /// </summary>
+    class AndroidBridge
+    {
+        private static int s_AndroidExtensionsExists = -1;
```
Comment (vincent-unity) [inline com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs:19]: Can we turn this into a enum like

private enum ExtensionState
{
     Unknow,
     Unavailable,
     Available
}

It's more clear than the C-style -1/0/1.

File: com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                get
+                {
+                    if (s_SdkRootPathProperty != null)
+                        return s_SdkRootPathProperty;
+                    s_SdkRootPathProperty = UnderlyingType.GetProperty("sdkRootPath");
+                    return s_SdkRootPathProperty;
+                }
+            }
+
+            public static string ndkRootPath
```
Comment (vincent-unity) [inline com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs:240]: NdkRootPath?

File: com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            {
+                get => (string)NdkRootPathProperty.GetValue(null);
+                set => NdkRootPathProperty.SetValue(null, value);
+            }
+
+            public static string sdkRootPath
```
Comment (vincent-unity) [inline com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs:246]: SdkRootPath?

**Review by vincent-unity — APPROVED** [review]

File: com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs (hunk 2 above)
Comment (todi1856) [reply com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs:240]: Sadly I want this to match Unity API property in this. I'll add comment

File: com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs (hunk 3 above)
Comment (todi1856) [reply com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs:246]: Sadly I want this to match Unity API property in this. I'll add comment

File: com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs (hunk 2 above)
Comment (vincent-unity) [reply com.unity.mobile.android-logcat/Editor/AndroidTools/AndroidBridge.cs:247]: Ah, I see

**Pull request merged**